hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/KeyManagerImpl.java

```diff
@@ -745,33 +745,34 @@ protected void refreshPipeline(List<OmKeyInfo> keyList) throws IOException {
     }
 
     Set<Long> containerIDs = new HashSet<>();
-    for (OmKeyInfo keyInfo : keyList) {
-      List<OmKeyLocationInfoGroup> locationInfoGroups =
-          keyInfo.getKeyLocationVersions();
-
-      for (OmKeyLocationInfoGroup key : locationInfoGroups) {
-        for (OmKeyLocationInfo k : key.getLocationList()) {
-          containerIDs.add(k.getContainerID());
-        }
-      }
-    }
+    keyList.forEach(
+        keyInfo -> keyInfo.getKeyLocationVersions().forEach(
+            key -> key.getLocationLists().forEach(
+                OmKeyLocationInfoList -> OmKeyLocationInfoList.forEach(
+                    k -> containerIDs.add(k.getContainerID())
```

Comment:
I see this is just using forEach instead of for, in addition to that with method usage.
Just curious, any reason to change this way?

Reply:
no... it's just a matter of code style. Having four layers of for loop looks bad, but for layers of forEach isn't any better.

Reply:
Yes. But if you feel that this way looks good, i am okay.
In general, if it is just a code refactor, it would be nice to avoid, so that we can know git history with annotation which has added this change. Now after this change is merged we get the last commit in git annotation.

Reply:
another thing i would also like to try is to use parallelStream() to parallelize this part. Need benchmark

Reply:
I don't expect that streams are worth it, even in parallel.  Benchmark is welcome in any case.  However, can we separate it from this fix, so that we can get this one in sooner rather than later?

Reply:
ok let's do that.